lib/src/bin/compile_sass.dart

```diff
@@ -30,63 +45,173 @@ const Map<String, sass.OutputStyle> outputStyleArgToOutputStyleValue = const {
+        print('$errorMessageHeading "$target" does not exist');
+        exitCode = 1;
+        break;
+      } else {
+        final targetRootDirName =
+            '${path.rootPrefix(target)}${path.split(target).first}';
+
+        if (srcRootDirName != null) {
+          if (targetRootDirName != srcRootDirName) {
+            print(
+                '$errorMessageHeading All targets must share the same root directory. Expected "$target" to exist within "$srcRootDirName".');
+            exitCode = 1;
+            break;
+          }
+        } else {
+          srcRootDirName = targetRootDirName;
+        }
+      }
+    }
+
+    return exitCode;
+  }
+}
+
+Future<Null> main(List<String> args) async {
+  taskTimer = new Stopwatch();
   final parser = new ArgParser()
     ..addMultiOption(outputStyleArg,
+        abbr: 's',
         help: 'The output style used to format the compiled CSS.',
         defaultsTo: outputStyleDefaultValue,
         splitCommas: true)
     ..addOption(expandedOutputStyleFileExtensionArg,
         help:
-            'The file extension that will be used for the CSS compiled using `expanded` outputStyle.',
+            'The file extension that will be used for the CSS compiled using \n`expanded` outputStyle.',
         defaultsTo: expandedOutputStyleFileExtensionDefaultValue)
     ..addOption(compressedOutputStyleFileExtensionArg,
         help:
-            'The file extension that will be used for the CSS compiled using `compressed` outputStyle unless more than one `--$outputStyleArg` is defined. When more than one outputStyle is used, the extension for compressed CSS will be `.min.css` no matter what.',
-        defaultsTo: compressedOutputStyleFileExtensionDefaultValue)
+            'The file extension that will be used for the CSS compiled using \n`compressed` outputStyle.\n'
+            '(defaults to $compressedOutputStyleFileExtensionDefaultValue, or .min.css\n'
+            ' if `--$outputStyleArg` contains more than one style)')
     ..addOption(sourceDirArg,
         help:
-            'The directory where the `.scss` files that you want to compile live. Defaults to $sourceDirDefaultValue, or the value of `--$outputDirArg`, if specified.')
+            'The directory where the `.scss` files that you want to compile live. \n(defaults to $sourceDirDefaultValue, or the value of `--$outputDirArg`, if specified.)')
     ..addOption(outputDirArg,
         help:
-            'The directory where the compiled CSS should go. Defaults to $outputDirDefaultValue, or the value of `--$sourceDirArg`, if specified.')
+            'The directory where the compiled CSS should go. \n(defaults to $outputDirDefaultValue, or the value of `--$sourceDirArg`, if specified.)')
+    ..addMultiOption(watchDirsArg,
+        splitCommas: true,
+        defaultsTo: const <String>[],
+        help:
+            'Directories that should be watched in addition to `sourceDir`. \nOnly valid with --watch.')
```

Comment:
What's the use case where additional watch directories are added that are separate from the source directory?

Reply:
@corwinsheahan-wf for instance if you have your top-level files in `lib`, but your partials are in `sass` instead of `lib/sass`.

Reply:
ah right right